pkg/ecdsa/retry.go

```diff
@@ -0,0 +1,62 @@
+package ecdsa
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+
+	"github.com/keep-network/keep-core/pkg/chain"
+)
+
+type byAddress []chain.Address
+
+func (ba byAddress) Len() int           { return len(ba) }
+func (ba byAddress) Swap(i, j int)      { ba[i], ba[j] = ba[j], ba[i] }
+func (ba byAddress) Less(i, j int) bool { return ba[i] < ba[j] }
+
+func EvaluateRetryParticipantsForSigning(
+	groupMembers []chain.Address,
+	seed int64,
+	retryCount uint,
+	quantity uint,
+) ([]chain.Address, error) {
+	if int(quantity) > len(groupMembers) {
+		return nil, fmt.Errorf(
+			"Asked for too many seats. %d seats were requested, but there are only %d available.",
```

Comment:
We typically start the error message with a lowercase due to the error wrapping. Also, we put format parameters into square braces:
```suggestion
			"asked for too many seats. [%d] seats were requested, but there are only [%d] available.",
```

Reply:
> due to the error wrapping

ahhh makes a bunch of sense

Reply:
[efc919723efa5b3cf9081bf691c916d102590e99](https://github.com/keep-network/keep-core/pull/3109/commits/efc919723efa5b3cf9081bf691c916d102590e99)